### PR TITLE
Associate users and opportunities

### PR DESCRIPTION
**Pull request by chrisrodz**
### User - Opportunity Association

@MrMaksimize please review.
- [x] Fix Choice Gen Structure
- [x] Review form gen structure for order loading -- fundme then admin
- [x] Test Choice Gen Structure
- [x] Fix Opp Query Form Gen Structure
- [x] Fundme Gender: [Male | Female | Both]
- [x] Fundme Age - Remove Any
- [x] Fundme -- What do you need this funding for, include other field.
- [x] Fundme - Are you investing your own money should control whether how much shows up
- [x] Fundme - Are you investing your own money should Not be required
- [x] Fundme - What is your business type - remove any
- [x] Fundme - What type of industry - when I click other, just leave other instead of repeating (https://www.pivotaltracker.com/story/show/74967564)
- [x] Fundme - How Many employees do you have - remove any
- [x] Fundme -- Years in Biz - Remove Any
- [x] Fundme -  Annual Rev - Remove Any (Possibly IDK).
## Association Stuff that's left -- @chrisrodz on u
- [x] Remove entrySave method
- [x] make save method work for both saves and updates.
- [x] Set the user on save method when opp is being saved.
- [x] fix tests
- [x] Clean up how things save on admin and in user -- see tests

Man this one is massive

### CODE REVIEW

File: controllers/admin.js [hunk 1]
```diff
@@ -101,6 +124,7 @@ function save(req, res) {
   var id = req.params.id || '';
   var Model = sequelize.isDefined(req.params.model) ? sequelize.model(req.params.model) : null;
   var instance = Model.buildFromAdminForm(req.body);
+  instance['user_id'] = 1;
```
Comment (MrMaksimize) [inline controllers/admin.js]: Req.user.id :)

Comment (MrMaksimize) [inline controllers/admin.js]: This should be a hook that runs every time a model is saved --- saved NEW

File: controllers/admin.js [hunk 2]
```diff
@@ -60,6 +62,27 @@ function list(req, res) {
...
+/**
+ * GET listAll
+ */
+function listAll(req, res) {
+  var render = _.extend(res.locals, {
+    model: req.params.model
+    //page: req.params.page || 0;
+  });
+
+  var Model = sequelize.model(render.model);
```
Comment (MrMaksimize) [inline controllers/admin.js]: wait -- why are you grabbing this twice (ln 82) and also you're using Model for an instance

File: controllers/admin.js (hunk 1 above)
Comment (chrisrodz) [inline controllers/admin.js]: Already discussed this with you :smile:

File: controllers/admin.js (hunk 2 above)
Comment (chrisrodz) [inline controllers/admin.js]: The listAll is for the future when we want to have a page that displays ALL opportunities, regardless of if they belong to a user or not.

You can comment it out for now, I just wanted to have it there already.

File: test/models.js [hunk 3]
```diff
@@ -143,6 +143,28 @@ describe('Opportunity Model', function() {
     });
   });
 
+  it('should create a new opportunity associated to a user by id', function(done) {
```
Comment (MrMaksimize) [inline test/models.js:98]: I have done a ton of stuff on this pr, but I'm super tired and don't wanna work on this damn test.  

Can you please update it so it does the following:
- Creates a user
- Creates a opp instance with an association to that user (see admin save)
- Makes sure that you can retrieve the model by that user (see admin list)
- Destroys the instance 
- Destroys the user.

Thanks @chrisrodz

**Comment by MrMaksimize** [comment]
Wait on 44 to get pulled

**Comment by chrisrodz** [comment]
@MrMaksimize I don't fully understand those last two checkboxes that have to do with tests. My current test suite passes.

**Comment by chrisrodz** [comment]
@MrMaksimize This is re-ready to get merged. We both have code here so I feel we should do it together. Let me know when.

File: controllers/admin.js [hunk 4]
```diff
@@ -100,29 +106,29 @@ function edit(req, res) {
...
-          req.flash('errors', errKey + ': ' + errDesc);
-      });
-      return res.json(err);
-      //return res.redirect(req.path);
-    }
 
-    instance.save().success(function(){
-      req.flash('info', instance.title + ' Successfully Added');
+  // If there is no id we are creating a new instance
+  if (id === '') {
```
Comment (MrMaksimize) [inline controllers/admin.js]: checking for '' and explicitly for a blank string could pass through undefined and null.  I'd say based on https://gist.github.com/MrMaksimize/00214273fd1a5c8156ab do somethinkg like

`if (!id || _.isEmpty(id)) {`

File: controllers/admin.js [hunk 5]
```diff
@@ -100,29 +106,29 @@ function edit(req, res) {
...
+    Model.createInstance(req.body).then(function(instance) {
+      return req.user.addOpportunity(instance);
```
Comment (MrMaksimize) [inline controllers/admin.js:113]: Yay promises!

File: migrations/20140711105052-add-more-multi-select-fields.js [hunk 6]
```diff
@@ -1,7 +1,9 @@
-var choicesList = require('../lib/options');
+var OptionsList = require('../lib/OptionsList');
+console.log('migration init options list.');
```
Comment (MrMaksimize) [inline migrations/20140711105052-add-more-multi-select-fields.js]: Shoot forgot about this console. log.  Kill this please :) ?

File: models/lib/modelUtils.js [hunk 7]
```diff
@@ -67,13 +70,31 @@ var classMethods = {
           element.value = valueSet.value;
           element.otherValue = valueSet.otherValue;
         }
-
+        if (key == 'gender') {
+          console.log('Gender Choices');
```
Comment (MrMaksimize) [inline models/lib/modelUtils.js]: And these console logs too pleeeeeese?

**Comment by MrMaksimize** [comment]
@chrisrodz awesome!!! just some super super simple things.  And lmk when you get to em and i'll merge tonight or early tomorrow :)

**Comment by MrMaksimize** [comment]
@chrisrodz made the fixes myself :)  Mostly mine anyways and didn't wanna have to wait.  As soon as travis clears i'll merge

**Comment by chrisrodz** [comment]
Awesome. I just got home from the South Bay. Once you merge I think I'll
fix up my other PR

On Monday, July 21, 2014, Maksim Pecherskiy notifications@github.com
wrote:

> @chrisrodz https://github.com/chrisrodz made the fixes myself :) Mostly
> mine anyways and didn't wanna have to wait. As soon as travis clears i'll
> merge
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CoquiCoders/bizwallet/pull/33#issuecomment-49702231.

## 

Best,
Christian A. Rodriguez

**Comment by MrMaksimize** [comment]
No worries :)  lol dude -- or you can do it tomorrow. Don't kill yourself over it :) Travis just cleared.  Chugga chugga chugga.  Here we go :)

**Pull request merged**